### DISCUSSION TITLE
How to check if a field is exist with a condition

### DISCUSSION THREAD

**Discussion by roylisto** [cue-lang/cue#2061] [Q&A] [answered]
Hi I want to check if a field is exist or not by using cue

is it possible to do something like this?

yaml-version?: 1

if yaml-version && yaml-version == 1 {

}

**Comment by slewiskelly** [comment, marked as answer]
If you want to check the presence of an optional field you can check that it is not bottom (`_|_`):

```cue
yaml_version?: int

exists: yaml_version != _|_ // false
```

If you want to use the value of an optional field, you always have to check against bottom first. A contrived, but similar, example being:

```cue
ok: bool | *false

if yaml_version != _|_ {
    ok: yaml_version == 1
}
```

The following will not work:

```cue
ok: yaml_version != _|_ && yaml_version == 1 // cannot reference optional field: yaml_version
```

**Reply by roylisto** [reply]
thanks a lot!

**Reply by roylisto** [reply]
a question on this line @slewiskelly 
```
if yaml_version != _|_ {
    ok: yaml_version == 1
}
```

so that means whether the condition is true or false the code will still be executed inside the block?

**Reply by myitcv** [reply]
That's a very good question. Those blocks are only included if the condition is true. 

See https://github.com/cue-lang/cue/issues/1730 for more details.

**Reply by roylisto** [reply]
and is this possible `ok: yaml_version == 1`?

**Comment by seh** [comment]
See #2097 for a related inquiry.

**Reply by myitcv** [reply]
See my reply in https://github.com/cue-lang/cue/discussions/2097#discussioncomment-4203756 for more detail on how this maps to the proposed builtins in #943.
